aws/retry/middleware.go

```diff
@@ -87,61 +83,103 @@ func (r Attempt) HandleFinalize(ctx context.Context, in smithymiddle.FinalizeInp
 			AttemptClockSkew: attemptClockSkew,
 		})
 
-		if attemptNum > 1 {
-			if rewindable, ok := in.Request.(interface{ RewindStream() error }); ok {
-				if err := rewindable.RewindStream(); err != nil {
-					return out, metadata, fmt.Errorf("failed to rewind transport stream for retry, %w", err)
-				}
-			}
+		var attemptResult AttemptResult
 
-			r.logf(logger, logging.Debug, "retrying request %s/%s, attempt %d", service, operation, attemptNum)
-		}
+		out, attemptResult, err = r.handleAttempt(attemptCtx, attemptInput, next)
 
-		out, metadata, reqErr := next.HandleFinalize(attemptCtx, attemptInput)
+		responseMetadata := awsmiddle.GetResponseMetadata(attemptResult.ResponseMetadata)
+		attemptClockSkew = responseMetadata.AttemptSkew
```

Comment:
Should `GetResponseMetadata` be remand to something more specific, like `GetResponseClockSckew` ?  `ResponseMetadata` seems out of place for what it is doing, and combined with the `AttemptResult(s)`